identity_credential/src/validator/credential_validator.rs

```diff
@@ -81,18 +81,15 @@ impl CredentialValidator {
 
   /// Validate that the [`Credential`] expires on or after the specified [`Timestamp`].
   pub fn check_expires_on_or_after<T>(credential: &Credential<T>, timestamp: Timestamp) -> ValidationUnitResult {
-    let is_ok = if let Some(expiration_date) = credential.expiration_date {
-      expiration_date >= timestamp
-    } else {
-      true
-    };
-    is_ok.then(|| ()).ok_or(ValidationError::ExpirationDate)
+    (credential.expiration_date.unwrap_or(timestamp) >= timestamp)
```

Comment:
Personal opinion: This is clever and concise, but I find the code less easily understandable now. I would prefer readability over number of lines of code.

Reply:
> I would prefer readability over number of lines of code.

I very much share this sentiment and I have to agree that although what I wrote is technically correct I consider it conceptually wrong. 

Would you find this easier to read?:
```
    let expiration_date = credential.expiration_date;
    (expiration_date.is_none() || expiration_date >= Some(timestamp))
      .then_some(())
      .ok_or(ValidationError::ExpirationDate)
```

Reply:
Yeah that's great. Don't forget to annotate the type on `expiration_date` :wink: